src/main/java/io/aiven/kafka/auth/AivenAclAuthorizerV2.java

```diff
@@ -68,19 +68,13 @@
     private File configFile;
     private AuditorAPI auditor;
     private boolean logDenials;
-    private final ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
-    private final WatchService watchService;
+    private ScheduledExecutorService scheduledExecutorService;
+    private WatchService watchService;
```

Comment:
Since it's not `final` anymore, I think it needs being `volatile` / `AtomicReference`. It's assigned in one thread and is used in another (via `subscribeToAclChanges`).